src/main/java/it/unitn/ds1/Replica.java

```diff
@@ -247,7 +308,46 @@ private void onReadMsg(ReadMsg msg) {
     }
 
     private void onHeartbeatMsg(HeartbeatMsg msg) {
+        if (this.isCoordinator) {
+            // Sends an heartbeat to all other replicas
+            this.multicast(msg);
```

Comment:
If all replicas send the heartbeat, and the coordinator sends a multicast, wouldn't it send too many messages?
We can just reply to the replica that requested the heartbeat.

Reply:
But the point of heartbeats is that they're sent by by the coordinator every once in a while and not request by replicas. So there's no risk to send of generating too many heartbeats.